src/GraphQL/Types/Scalars/ShortGraphType.cs

```diff
@@ -12,25 +12,56 @@ public class ShortGraphType : ScalarGraphType
         /// <inheritdoc/>
         public override object ParseLiteral(IValue value) => value switch
         {
-            IntValue intValue => short.MinValue <= intValue.Value && intValue.Value <= short.MaxValue ? (short?)intValue.Value : null,
-            LongValue longValue => short.MinValue <= longValue.Value && longValue.Value <= short.MaxValue ? (short?)longValue.Value : null,
-            BigIntValue bigIntValue => short.MinValue <= bigIntValue.Value && bigIntValue.Value <= short.MaxValue ? (short?)bigIntValue.Value : null,
-            _ => null
+            IntValue intValue => checked((short)intValue.Value),
+            LongValue longValue => checked((short)longValue.Value),
+            BigIntValue bigIntValue => checked((short)bigIntValue.Value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
         };
 
-        /// <inheritdoc/>
-        public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(short));
-
         /// <inheritdoc/>
         public override bool CanParseLiteral(IValue value) => value switch
         {
             IntValue intValue => short.MinValue <= intValue.Value && intValue.Value <= short.MaxValue,
             LongValue longValue => short.MinValue <= longValue.Value && longValue.Value <= short.MaxValue,
             BigIntValue bigIntValue => short.MinValue <= bigIntValue.Value && bigIntValue.Value <= short.MaxValue,
+            NullValue _ => true,
             _ => false
         };
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new IntValue(Convert.ToInt16(value));
+        public override object ParseValue(object value) => value switch
+        {
+            short _ => value,
+            null => null,
+            int i => checked((short)i),
+            sbyte sb => checked((short)sb),
+            byte b => checked((short)b),
+            ushort us => checked((short)us),
+            uint ui => checked((short)ui),
+            long l => checked((short)l),
+            ulong ul => checked((short)ul),
+            BigInteger bi => (short)bi,
```

Comment:
Why without `checked`? Other widening conversions like ` byte b => checked((short)b),` have `checked`.

Reply:
Sure, I'll add it in.  FYI, it's not a native conversion, so checked/unchecked does not have any effect (tested & confirmed).  It simply calls the BigInteger explicit cast operator.

Reply:
We also have tests that confirm all these casts throw exceptions properly, for every type, to every type.

Reply:
Good. I thought so. Just the reader will arise less questions if everywhere will be equally.